### ISSUE TITLE
InvalidOperationError on trying to select array with pl.lit

### ISSUE THREAD

**Issue by Kryzpoo** [pola-rs/polars#18709]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
df1 = pl.DataFrame(dict(id=[1, 2, 3]))
df1.select(id=pl.col('id'), aggs=pl.lit([]))
```

### Log output

```shell
Traceback (most recent call last):
  File "/root/.pycharm_helpers/pydevd_asyncio/pydevd_asyncio_utils.py", line 117, in _exec_async_code
    result = func()
             ^^^^^^
  File "<input>", line 1, in <module>
  File "/usr/local/lib/python3.11/site-packages/polars/dataframe/frame.py", line 8968, in select
    return self.lazy().select(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/polars/lazyframe/frame.py", line 2034, in collect
    return wrap_df(ldf.collect(callback))
                   ^^^^^^^^^^^^^^^^^^^^^
polars.exceptions.InvalidOperationError: Series: aggs, length 1 doesn't match the DataFrame height of 3
If you want this Series to be broadcasted, ensure it is a scalar (for instance by adding '.first()').
```

### Issue description

Problem appeared in polars 1.7: when i'm trying to append to DF a constant array column

### Expected behavior

In 1.6, i would get DF with constant array column:
```
shape: (3, 2)
┌─────┬──────┐
│ id  ┆ aggs │
│ --- ┆ ---  │
│ i64 ┆ list  │
╞═════╪══════╡
│ 1   ┆[]    │
│ 2   ┆ []    │
│ 3   ┆ []    │
└─────┴──────┘
```

### Installed versions

<details>

```
pl.show_versions()
--------Version info---------
Polars:              1.7.0
Index type:          UInt32
Platform:            Linux-5.15.49-linuxkit-pr-aarch64-with-glibc2.28
Python:              3.11.4 (main, Jun 13 2023, 11:34:58) [GCC 8.3.0]

```

</details>

**Comment by cmdlineluser** [comment]
- https://github.com/pola-rs/polars/issues/18686

**Comment by ritchie46** [comment]
fixed by #18707

I shall patch
